### PR TITLE
deployer: skip rollback on all-at-once if old jobs failed to stop in timely fashion

### LINKED ISSUE #2162: Deployer should not rollback if jobs fail to stop in a timely manner
If we're doing an `all-at-once` deploy and the jobs fail to stop quickly, the deploy is rolled back, which doesn't make a ton of sense.

### PR DESCRIPTION
**Pull request by imjorge**
Closes #2162

### CODE REVIEW

File: controller/worker/deployment/all_at_once.go [hunk 1]
```diff
@@ -72,7 +72,7 @@ func (d *DeployJob) deployAllAtOnce() error {
 		log.Info("waiting for job events", "expected", expected)
 		if err := d.waitForJobEvents(d.OldReleaseID, expected, log); err != nil {
 			log.Error("error waiting for job events", "err", err)
-			return err
+			return ErrSkipRollback{err.Error()}
```
Comment (lmars) [inline controller/worker/deployment/all_at_once.go]: Probably worth a comment (something along the lines of [this comment](https://github.com/flynn/flynn/issues/2162#issuecomment-157908326)).

Comment (imjorge) [inline controller/worker/deployment/all_at_once.go]: @lmars done.

**Comment by lmars** [comment]
LGTM

**Comment by lmars** [comment]
Thanks!

**Pull request merged**